src/emulator/system.c

```diff
@@ -0,0 +1,1801 @@
+            }
+            BUFFER_GET(pBuffer, 0) = -1;
+            if (pSystem->eTypeROM == 'NMQE') {
+                if (!cpuSetCodeHack(pCPU, 0x8005E98C, 0x1040FFFF, -1)) {
+                    return false;
+                }
+                if (!cpuSetCodeHack(pCPU, 0x8005F2D8, 0x1440FFFD, -1)) {
+                    return false;
+                }
+            } else if (pSystem->eTypeROM == 'NMQJ') {
+                if (!cpuSetCodeHack(pCPU, 0x8005E63C, 0x1040FFFF, -1)) {
+                    return false;
+                }
+                if (!cpuSetCodeHack(pCPU, 0x8005EF88, 0x1440FFFD, -1)) {
+                    return false;
+                }
+            }
+            break;
+        case 'NPOE':
+        case 'NPOP':
+        case 'NPOJ':
+            storageDevice = SOT_FLASH;
+            gSystemRomConfigurationList.storageDevice = SOT_RAM;
+            break;
+        case 'NQKJ':
+        case 'NQKP':
+        case 'NQKE':
+            if (!cpuSetCodeHack(pCPU, 0x8004989C, 0x1459FFFB, 0)) {
+                return false;
+            }
+            if (!cpuSetCodeHack(pCPU, 0x80049FF0, 0x1608FFFB, 0)) {
+                return false;
+            }
+            if (!cpuSetCodeHack(pCPU, 0x8004A384, 0x15E0FFFB, 0)) {
+                return false;
+            }
+            if (!cpuSetCodeHack(pCPU, 0x8004A97C, 0x15E0FFFB, 0)) {
+                return false;
+            }
+            if (!cpuSetCodeHack(pCPU, 0x80048FF8, 0x1000FFFD, 0x1000FFFF)) {
+                return false;
+            }
+            break;
+        case 'NGUJ':
+        case 'NGUP':
+        case 'NGUE':
+            gSystemRomConfigurationList.storageDevice = SOT_RSP;
+            pArgument = 0x1000;
+            storageDevice = SOT_FLASH;
+            if (!ramGetBuffer(SYSTEM_RAM(gpSystem), &pBuffer2, 0x300, NULL)) {
+                return false;
+            }
+            BUFFER_GET(pBuffer2, 4) = 0x17D6;
+            if (!cpuSetCodeHack(pCPU, 0x80025D30, 0x3C018006, -1)) {
+                return false;
+            }
+            break;
+        case 'NSQP':
+        case 'NSQJ':
+        case 'NSQE':
+            if (!ramGetBuffer(SYSTEM_RAM(gpSystem), &pBuffer2, 0x300, NULL)) {
+                return false;
+            }
+            BUFFER_GET(pBuffer2, 4) = 0x17D6;
+            break;
+        case 'NOBJ':
+        case 'NOBP':
+        case 'NOBE':
+            pArgument = 0x8000;
+            gSystemRomConfigurationList.storageDevice = SOT_PIF;
+            storageDevice = SOT_SRAM;
+            if (!ramGetBuffer(SYSTEM_RAM(gpSystem), &pBuffer2, 0x300, NULL)) {
+                return false;
+            }
+            BUFFER_GET(pBuffer2, 4) = 0x17D6;
+            break;
+        case 'NRXP':
+        case 'NRXJ':
+        case 'NRXE':
+            if (!ramGetBuffer(SYSTEM_RAM(gpSystem), &pBuffer2, 0x300, NULL)) {
+                return false;
+            }
+            BUFFER_GET(pBuffer2, 4) = 0x17D6;
```

Comment:
I find it a bit weird that BUFFER_GET() is used for setting too. Does it work if `pBuffer` is an s32* and we just do `pBuffer2[4]` etc here?

Reply:
it worked for both buffers